src/controllers/user.controller.js

```diff
@@ -128,35 +128,48 @@ export class UserController {
     }
   }
 
-    static async getProfile(req, res){
-        try {
-            const user = await UserService.getUserById(req.user.id);
-            return res.status(200).json(user);
-        }catch (err){
-            return res.status(500).json({
-                error: err.message,
-                message: 'Failed to get user profile',
-            });
-        }
+  static async getProfile(req, res) {
+    try {
+      const user = await UserService.getUserById(req.user.id);
+      return res.status(200).json(user);
+    } catch (err) {
+      return res.status(500).json({
+        error: err.message,
+        message: 'Failed to get user profile',
+      });
     }
-    static async updateProfile(req, res){
-        try{
-            let payload = {
-                ...req.body,
-                billingAddress:JSON.parse(req.body.billingAddress) || {},
-                birthdate:new Date(req.body.birthdate || ''),
-            };
-            if(req.files?.avatar){
-                const {url} = await uploadPhoto(req,res,req.files.avatar);
-                payload['avatar'] = url;
-            }
-            await UserService.updateUser(req.user.id, payload);
-            return res.status(200).json({ message: 'updated successful'});
-        }catch (err){
-            return res.status(500).json({
-                error: err.message,
-                message: 'Failed to update user profile',
-            });
-        }
+  }
+  static async updateProfile(req, res) {
+    try {
+      let payload = {
+        ...req.body,
+        billingAddress: JSON.parse(req.body.billingAddress) || {},
+        birthdate: new Date(req.body.birthdate || ''),
+      };
+      if (req.files?.avatar) {
+        const { url } = await uploadPhoto(req, res, req.files.avatar);
+        payload['avatar'] = url;
+      }
+      await UserService.updateUser(req.user.id, payload);
+      return res.status(200).json({ message: 'updated successful' });
+    } catch (err) {
+      return res.status(500).json({
+        error: err.message,
+        message: 'Failed to update user profile',
+      });
+    }
+  }
+  static async updateStatus(req, res) {
+    try {
+      const status = req.body.status;
+      await UserService.updateUser({ status }, req.user.id);
+      return res.status(200).json({ message: 'Status updated successfully' });
+
+    } catch (error) {
+      return res.status(500).json({
+        error: error.message,
+        message: 'Failed to update status',
+      });
```

Comment:
You have to rename the method name and the success message, because this is about to disable account not update status